crates/ironrdp-web/src/audio.rs

```diff
@@ -0,0 +1,773 @@
+        if let Err(e) = document.add_event_listener_with_callback("click", callback_ref) {
+            warn!("Failed to add click listener for audio activation: {e:?}");
+        }
+        if let Err(e) = document.add_event_listener_with_callback("keydown", callback_ref) {
+            warn!("Failed to add keydown listener for audio activation: {e:?}");
+        }
+        if let Err(e) = document.add_event_listener_with_callback("touchstart", callback_ref) {
+            warn!("Failed to add touchstart listener for audio activation: {e:?}");
+        }
+
+        info!("Audio gesture activation listeners registered (click, keydown, touchstart)");
+
+        // Store the closure to prevent it from being dropped
+        *activation_handler.borrow_mut() = Some(activate_closure);
+    }
+}
+
+impl AudioQueue {
+    /// Enqueue an audio buffer for playback through the specified gain node
+    fn enqueue_audio(&mut self, buffer: AudioBuffer, gain_node: &GainNode) -> Result<(), IronError> {
+        // Validate audio context state
+        match self.context.state() {
+            web_sys::AudioContextState::Running => {}
+            web_sys::AudioContextState::Suspended => {
+                return Err(anyhow::Error::msg("Audio context suspended - user gesture required").into());
+            }
+            web_sys::AudioContextState::Closed => {
+                return Err(anyhow::Error::msg("Audio context is closed").into());
+            }
+            _ => {
+                return Err(anyhow::Error::msg("Audio context in unknown state").into());
+            }
+        }
+        let source = self
+            .context
+            .create_buffer_source()
+            .map_err(|e| anyhow::Error::msg(format!("Failed to create buffer source: {e:?}")))?;
+
+        source.set_buffer(Some(&buffer));
+
+        // Connect through the gain node for proper volume control
+        source
+            .connect_with_audio_node(gain_node)
+            .map_err(|e| anyhow::Error::msg(format!("Failed to connect buffer source to gain node: {e:?}")))?;
+
+        // Use the audio context's actual current time with proper scheduling
+        let context_time = self.context.current_time();
+        let start_time = context_time.max(self.current_time);
+
+        source
+            .start_with_when(start_time)
+            .map_err(|e| anyhow::Error::msg(format!("Failed to start audio playback: {e:?}")))?;
+
+        // Update tracking time for next buffer
+        self.current_time = start_time + buffer.duration();
+
+        info!(
+            "Audio buffer scheduled: duration={:.3}s, start_time={:.3}s, next_time={:.3}s, context_time={:.3}s",
+            buffer.duration(),
+            start_time,
+            self.current_time,
+            context_time
+        );
+
+        Ok(())
+    }
+}
+
+// SAFETY: In WebAssembly single-threaded environment, Send is safe for WebAudioBackend
+unsafe impl Send for WebAudioBackend {}
```

Comment:
nitpick: Strictly speaking, Send is really to communicate that the type is thread-safe, that it is safe to send it to another thread, and it’s obviously **not the case** for WebAudioBackend (using `Rc` instead of `Arc`, etc).
So we should rather say this is a workaround that is not compromising the memory safety of the program, because as far as we are concerned for _this_ crate which is compiled into a WASM module and run in a single-threaded environment, `WebAudioBackend` is never actually used in a multi-threaded context.
The real fix would be to relax the Send bound somewhere, but I’m okay with the workaround for the purpose of getting stuff done.